src/xdp/trn_transit_xdp.c

```diff
@@ -454,6 +457,26 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 
 	__be64 tunnel_id = trn_vni_to_tunnel_id(pkt->geneve->vni);
 
+	/*
+	* If target ip is towards a virtual subnet, send it up to user space
+	*/
+	struct network_key_t nkey;
+	struct network_t *net;
+	nkey.prefixlen = 96;
+
+	// "__u32 nip[3]" has 12 bytes altogether
+	// tunnel_id is bit-wise 64 bit and occupies the first 8 bytes
+	// tip (target inner ip) occupies the rest 4 bytes
+	__builtin_memcpy(&nkey.nip[0], &tunnel_id, sizeof(tunnel_id));
+	nkey.nip[2] = pkt->inner_ip->daddr;
+	net = bpf_map_lookup_elem(&networks_map, &nkey);
+	// send pkt to user space if this is the edge gateway and dest inner ip belongs to a virtual subnet
+	if ( net && net->virtual && pkt->ip->daddr == net->cluster_gateway) {
+		bpf_debug("[Transit:%d:0x%x] Edge gateway\n", __LINE__, bpf_ntohl(pkt->inner_ip->daddr));
+		// Rewrite RTS and update cache
```

Comment:
this comment is probably copy pasted. the function of XDP_PASS here is to send pkt up to user space.

Reply:
Updated